File: android-app/app/src/main/java/com/lasohealth/android/core/model/HealthCategory.kt
package com.lasohealth.android.core.model

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.DirectionsRun
import androidx.compose.material.icons.automirrored.filled.DirectionsWalk
import androidx.compose.material.icons.filled.Favorite
import androidx.compose.material.icons.filled.Hearing
import androidx.compose.material.icons.filled.Hotel
import androidx.compose.material.icons.filled.Air
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.Psychology
import androidx.compose.material.icons.filled.Restaurant
import androidx.compose.material.icons.filled.SelfImprovement
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import com.lasohealth.android.ui.theme.CategoryActivity
import com.lasohealth.android.ui.theme.CategoryBody
import com.lasohealth.android.ui.theme.CategoryHearing
import com.lasohealth.android.ui.theme.CategoryHeart
import com.lasohealth.android.ui.theme.CategoryMindfulness
import com.lasohealth.android.ui.theme.CategoryMobility
import com.lasohealth.android.ui.theme.CategoryNutrition
import com.lasohealth.android.ui.theme.CategoryRespiratory
import com.lasohealth.android.ui.theme.CategorySleep

// iOS SF Symbol → Material Icon mapping:
// heart.fill → Favorite, bed.double.fill → Hotel, figure.run → DirectionsRun,
// figure.stand → SelfImprovement, lungs.fill → Air (use Psychology as closest),
// brain.head.profile → Psychology, figure.walk.motion → DirectionsWalk,
// fork.knife → Restaurant, ear.fill → Hearing

enum class HealthCategory(
    val displayName: String,
    val shortName: String,
    val accentColor: Color,
    val icon: ImageVector,
) {
    HEART("Heart & Cardio", "Heart", CategoryHeart, Icons.Filled.Favorite),
    SLEEP("Sleep", "Sleep", CategorySleep, Icons.Filled.Hotel),
    ACTIVITY("Activity", "Activity", CategoryActivity, Icons.AutoMirrored.Filled.DirectionsRun),
    BODY("Body & Vitals", "Body", CategoryBody, Icons.Filled.Person),
    RESPIRATORY("Respiratory", "Lungs", CategoryRespiratory, Icons.Filled.Air),
    MINDFULNESS("Mindfulness", "Mind", CategoryMindfulness, Icons.Filled.SelfImprovement),
    MOBILITY("Mobility", "Mobility", CategoryMobility, Icons.AutoMirrored.Filled.DirectionsWalk),
    NUTRITION("Nutrition", "Nutrition", CategoryNutrition, Icons.Filled.Restaurant),
    HEARING("Hearing", "Hearing", CategoryHearing, Icons.Filled.Hearing),
}

File: Core/Models/HealthCategory.swift
import Foundation
import SwiftUI

/// Top-level health categories grouping related metrics
enum HealthCategory: String, CaseIterable, Identifiable, Codable {
    var id: String { rawValue }

    case heart
    case sleep
    case activity
    case body
    case respiratory
    case mindfulness
    case mobility
    case nutrition
    case hearing

    var displayName: String {
        switch self {
        case .heart: return "Heart & Cardio"
        case .sleep: return "Sleep"
        case .activity: return "Activity"
        case .body: return "Body & Vitals"
        case .respiratory: return "Respiratory"
        case .mindfulness: return "Mindfulness"
        case .mobility: return "Mobility"
        case .nutrition: return "Nutrition"
        case .hearing: return "Hearing"
        }
    }

    var systemImageName: String {
        switch self {
        case .heart: return "heart.fill"
        case .sleep: return "bed.double.fill"
        case .activity: return "figure.run"
        case .body: return "figure.stand"
        case .respiratory: return "lungs.fill"
        case .mindfulness: return "brain.head.profile"
        case .mobility: return "figure.walk.motion"
        case .nutrition: return "fork.knife"
        case .hearing: return "ear.fill"
        }
    }

    var color: Color {
        switch self {
        case .heart: return AppColour.categoryHeart
        case .sleep: return AppColour.categorySleep
        case .activity: return AppColour.categoryActivity
        case .body: return AppColour.primary
        case .respiratory: return AppColour.accent
        case .mindfulness: return AppColour.categoryStress
        case .mobility: return AppColour.success
        case .nutrition: return AppColour.warning
        case .hearing: return AppColour.stateDefault
        }
    }

    var shortName: String {
        switch self {
        case .heart: return "Heart"
        case .sleep: return "Sleep"
        case .activity: return "Activity"
        case .body: return "Body"
        case .respiratory: return "Lungs"
        case .mindfulness: return "Mind"
        case .mobility: return "Mobility"
        case .nutrition: return "Nutrition"
        case .hearing: return "Hearing"
        }
    }

    var metrics: [HealthMetric] {
        HealthMetric.allCases.filter { $0.category == self }
    }
}
